cirq/ops/controlled_operation.py

```diff
@@ -11,55 +11,69 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Union, Any, Optional
+from typing import Union, Any, Optional, Iterable, List, Sequence
 
 import numpy as np
 
 from cirq import protocols, linalg, value
-from cirq.ops import raw_types
+from cirq.ops import raw_types, gate_operation
 from cirq.type_workarounds import NotImplementedType
 
 
 @value.value_equality
 class ControlledOperation(raw_types.Operation):
+    def __new__(cls,
+                controls: Sequence[raw_types.Qid],
+                sub_operation: raw_types.Operation):
+        """Auto-flatten nested controlled operations."""
+        if isinstance(sub_operation, ControlledOperation):
+            return ControlledOperation(
+                list(controls) + sub_operation.controls,
+                sub_operation.sub_operation)
+        return super().__new__(cls)
+
     def __init__(self,
-                 control: raw_types.Qid,
+                 controls: Sequence[raw_types.Qid],
                  sub_operation: raw_types.Operation):
-        self.control = control
+        self.controls = list(controls)
         self.sub_operation = sub_operation
 
     @property
     def qubits(self):
-        return (self.control,) + self.sub_operation.qubits
+        return tuple(self.controls) + self.sub_operation.qubits
```

Comment:
Just `self.controls + self.sub_operation.qubits` once `self.controls` is a tuple.